flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/cleanup/DefaultResourceCleaner.java

```
@@ -166,21 +188,49 @@ private DefaultResourceCleaner(
         mainThreadExecutor.assertRunningInMainThread();
 
         CompletableFuture<Void> cleanupFuture = FutureUtils.completedVoidFuture();
-        for (T cleanup : prioritizedCleanup) {
-            cleanupFuture = cleanupFuture.thenCompose(ignoredValue -> withRetry(jobId, cleanup));
+        for (Map.Entry<String, T> cleanup : prioritizedCleanup.entrySet()) {
+            cleanupFuture =
+                    cleanupFuture.thenCompose(
+                            ignoredValue -> withRetry(jobId, cleanup.getKey(), cleanup.getValue()));
         }
 
         return cleanupFuture.thenCompose(
                 ignoredValue ->
                         FutureUtils.completeAll(
-                                regularCleanup.stream()
-                                        .map(cleanup -> withRetry(jobId, cleanup))
+                                regularCleanup.entrySet().stream()
+                                        .map(
+                                                cleanupEntry ->
+                                                        withRetry(
+                                                                jobId,
+                                                                cleanupEntry.getKey(),
+                                                                cleanupEntry.getValue()))
                                         .collect(Collectors.toList())));
     }
 
-    private CompletableFuture<Void> withRetry(JobID jobId, T cleanup) {
+    private CompletableFuture<Void> withRetry(JobID jobId, String label, T cleanup) {
         return FutureUtils.retryWithDelay(
-                () -> cleanupFn.cleanupAsync(cleanup, jobId, cleanupExecutor),
+                () ->
+                        cleanupFn
+                                .cleanupAsync(cleanup, jobId, cleanupExecutor)
+                                .whenComplete(
```

Isn't this swallowing the exception?